### DISCUSSION TITLE
Keep connection alive to keep sending messages

### DISCUSSION THREAD

**Discussion by ncastro** [khoih-prog/WebSockets2_Generic#31] [Q&A]
Hi there.

First of all thank you for this nice library.

I'm programing on two arduino nano 33 iot, one is the server and the other the client, and I'm finding it difficult to keep a connection alive for an undefined period of time so I can send messages back and forward at a high speed.

It seems to me that the client on the server side stops being available but I don't know why. 

Can you provide an example of a server and client that keep the connection alive for as long as it needs instead of making the handshake all over again.

Thanks in advance.

**Comment by khoih-prog** [comment]
Thanks for your interests and usage of the library.

> It seems to me that the client on the server side stops being available but I don't know why.

As you know, it's very difficult and even impossible to guess what's wrong with this kind of details, even with the best crystal balls. The majority of the issues lies in your code / use-case. To be able to get any help from the community, you have to create an [MRE](https://stackoverflow.com/help/minimal-reproducible-example), then hopefully someone can spend some free time to investigate the issue.

> Can you provide an example of a server and client that keep the connection alive for as long as it needs instead of making the handshake all over again.

Sorry I'm quite busy with other more pressing works, and can't hep you. If you can provide the above-mentioned MRE, someone or I'll possibly spend some time to have a detailed look at hopefully have some better idea.

One more possiblility is port your code into ESP32/ESP8266, then test to see if the issue persists, if yes, post an issue on [ArduinoWebsockets Library](https://github.com/gilmaimon/ArduinoWebsockets).

Good Luck,
